packages/react-dom-bindings/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -2497,98 +2618,246 @@ export function writeCompletedBoundaryInstruction(
+      writeChunk(destination, formattedContentID);
+      writeChunk(destination, completeBoundaryOrContainerScript3);
+      return writeChunkAndReturn(
+        destination,
+        completeBoundaryOrContainerScriptEnd,
+      );
     }
   } else {
     if (enableFloat && hasStyleDependencies) {
       writeChunk(destination, completeBoundaryWithStylesData1);
+      writeChunk(destination, boundaryID);
+      writeChunk(destination, completeBoundaryOrContainerData2);
+      writeChunk(destination, responseState.segmentPrefix);
+      writeChunk(destination, formattedContentID);
+      writeChunk(destination, completeBoundaryOrContainerData3);
+      writeStyleResourceDependenciesInAttr(destination, boundaryResources);
+      return writeChunkAndReturn(
+        destination,
+        completeBoundaryOrContainerDataEmptyEnd,
+      );
     } else {
       writeChunk(destination, completeBoundaryData1);
+      writeChunk(destination, boundaryID);
+      writeChunk(destination, completeBoundaryOrContainerData2);
+      writeChunk(destination, responseState.segmentPrefix);
+      writeChunk(destination, formattedContentID);
+      return writeChunkAndReturn(
+        destination,
+        completeBoundaryOrContainerDataEmptyEnd,
+      );
     }
   }
+}
 
+export function writeCompletedRootBoundaryInstruction(
+  destination: Destination,
+  responseState: ResponseState,
+  boundaryID: SuspenseBoundaryID,
+  contentSegmentID: number,
+  boundaryResources: BoundaryResources,
+): boolean {
   if (boundaryID === null) {
     throw new Error(
       'An ID must have been assigned before we can complete the boundary.',
     );
   }
-
-  // Write function arguments, which are string and array literals
+  let hasStyleDependencies;
+  if (enableFloat) {
+    hasStyleDependencies = hasStyleResourceDependencies(boundaryResources);
+  }
   const formattedContentID = stringToChunk(contentSegmentID.toString(16));
-  writeChunk(destination, boundaryID);
+  const scriptFormat =
+    !enableFizzExternalRuntime ||
+    responseState.streamingFormat === ScriptStreamingFormat;
+  let r = true;
   if (scriptFormat) {
-    writeChunk(destination, completeBoundaryScript2);
-  } else {
-    writeChunk(destination, completeBoundaryData2);
-  }
-  writeChunk(destination, responseState.segmentPrefix);
-  writeChunk(destination, formattedContentID);
-  if (enableFloat && hasStyleDependencies) {
-    // Script and data writers must format this differently:
-    //  - script writer emits an array literal, whose string elements are
-    //    escaped for javascript  e.g. ["A", "B"]
-    //  - data writer emits a string literal, which is escaped as html
-    //    e.g. [&#34;A&#34;, &#34;B&#34;]
-    if (scriptFormat) {
-      writeChunk(destination, completeBoundaryScript3a);
-      // boundaryResources encodes an array literal
+    if (enableFloat && hasStyleDependencies) {
+      // We emit the bootstrap chunks into a template container with an id
```

Comment:
This is a clever solution. Curious - was the decision to emit bootstrap container as `bs:[sid]` to support multiple React SSR roots within a page?

Reply:
It does potentially enable that however it's unclear how there would be more than one root streaming into the same document at the same time. I'm sure someone could hack it to make that happen but its not something I was trying to explicitly support. I used the sid primarily to add some entropy to the id to avoid colliding with someone else's `bs`. the sid has the identifier prefix so in theory you can customize it not to collide if you really needed to use `bs` somewhere else